### PR TITLE
fix to sanFqdn lint and 3 new tests

### PR DESCRIPTION
**Pull request by zhengping12**
Fixed the issue that the lint fails wildcard fqdn DNS names.

### CODE REVIEW

**Comment by kumarde** [comment]
Where else is util.fqdn used? It is possible that while wildcard domains are allowed for DNSNames in SAN, they may not be allowed elsewhere. Could you check that and make any changes to other lints if they need to be changed?

**Comment by zhengping12** [comment]
It is used also in IAN, the exact same rule apply. I'll add equivalent tests for the IAN lint.

**Comment by zhengping12** [comment]
@kumarde added tests for ext_ian_uri_host_not_fqdn_or_ip and ext_san_uri_host_not_fqdn_or_ip.
I believe that's all there is relevant to fqdn.

**Comment by zakird** [comment]
I know this is going to sound pedantic, but can you please capitalize abbreviations correctly? E.g., `Fqdn` -> `FQDN`, `Uri` -> `URI`, `Dns` -> `DNS`, `San`- > `SAN`.

**Comment by zhengping12** [comment]
@zakird No worries! I didn't know all capitalized would work with the camelcase scheme. But they are fixed now.

File: lints/lint_ext_ian_dns_not_ia5_string_test.go [hunk 1]
```diff
@@ -5,8 +5,8 @@ import (
 	"testing"
 )
 
-func TestIanDnsIa5(t *testing.T) {
-	inputPath := "../testlint/testCerts/ianDnsIa5.cer"
```
Comment (zakird) [inline lints/lint_ext_ian_dns_not_ia5_string_test.go]: Should be `IA5String`

File: lints/lint_ext_ian_no_entries.go [hunk 2]
```diff
@@ -18,21 +18,21 @@ import (
...
-func (l *ianNoEntry) CheckApplies(c *x509.Certificate) bool {
+func (l *IANNoEntry) CheckApplies(c *x509.Certificate) bool {
 	return util.IsExtInCert(c, util.IssuerANOID)
 }
 
-func (l *ianNoEntry) RunTest(c *x509.Certificate) (ResultStruct, error) {
-	ian := util.GetExtFromCert(c, util.IssuerANOID)
-	if (ian.Value)[1] == 0 {
+func (l *IANNoEntry) RunTest(c *x509.Certificate) (ResultStruct, error) {
+	IAN := util.GetExtFromCert(c, util.IssuerANOID)
```
Comment (zakird) [inline lints/lint_ext_ian_no_entries.go]: This should probably be lowercase as it is not a constant.

File: lints/lint_ext_ian_space_dns_name_test.go [hunk 3]
```diff
@@ -18,8 +18,8 @@ func TestIanEmptyDns(t *testing.T) {
 	}
 }
 
-func TestIanNotEmptyDns(t *testing.T) {
-	inputPath := "../testlint/testCerts/ianNonEmptyDns.cer"
+func TestIanNotEmptyDNS(t *testing.T) {
```
Comment (zakird) [inline lints/lint_ext_ian_space_dns_name_test.go]: This seems like it should be `IAN`.

File: lints/lint_ext_ian_uri_format_invalid_test.go [hunk 4]
```diff
@@ -18,8 +18,8 @@ func TestIanUriValid(t *testing.T) {
 	}
 }
 
-func TestIanUriNoScheme(t *testing.T) {
-	inputPath := "../testlint/testCerts/ianURINoScheme.cer"
+func TestIanURINoScheme(t *testing.T) {
```
Comment (zakird) [inline lints/lint_ext_ian_uri_format_invalid_test.go]: again seems like this is incorrect.

File: lints/lint_ext_ian_uri_host_not_fqdn_or_ip.go [hunk 5]
```diff
@@ -18,24 +18,24 @@ import (
 	"github.com/zmap/zlint/util"
 )
 
-type ianUriFqdnOrIp struct {
+type IANURIFQDNOrIp struct {
 	// Internal data here
 }
 
-func (l *ianUriFqdnOrIp) Initialize() error {
+func (l *IANURIFQDNOrIp) Initialize() error {
```
Comment (zakird) [inline lints/lint_ext_ian_uri_host_not_fqdn_or_ip.go]: Shouldn't `IP` be capitalized?

File: lints/lint_ext_san_edi_party_name_present_test.go [hunk 6]
```diff
@@ -18,8 +18,8 @@ func TestSanEdiPartyPresent(t *testing.T) {
 	}
 }
 
-func TestSanEdiPartyMissing(t *testing.T) {
-	inputPath := "../testlint/testCerts/sanOtherName.cer"
+func TestSANEdiPartyMissing(t *testing.T) {
```
Comment (zakird) [inline lints/lint_ext_san_edi_party_name_present_test.go]: EDI should be capitalized.

File: lints/lint_ext_san_critical_with_subject_dn.go [hunk 7]
```diff
@@ -26,12 +26,12 @@ func (l *ExtSANCriticalWithSubjectDN) Initialize() error {
...
 
 func (l *ExtSANCriticalWithSubjectDN) CheckApplies(cert *x509.Certificate) bool {
-	return util.IsExtInCert(cert, util.SanOID)
+	return util.IsExtInCert(cert, util.SANOID)
 }
 
 func (l *ExtSANCriticalWithSubjectDN) RunTest(cert *x509.Certificate) (ResultStruct, error) {
-	san := util.GetExtFromCert(cert, util.SanOID)
-	if san.Critical && util.NotAllNameFieldsAreEmpty(&cert.Subject) {
+	SAN := util.GetExtFromCert(cert, util.SANOID)
```
Comment (zakird) [inline lints/lint_ext_san_critical_with_subject_dn.go]: should be lowercase

File: lints/lint_ian_dns_name_includes_null_char.go [hunk 8]
```diff
@@ -1,4 +1,4 @@
-// lint_ian_dns_name_includes_null_char.go
+// lint_IAN_dns_name_includes_null_char.go
```
Comment (zakird) [inline lints/lint_ian_dns_name_includes_null_char.go]: probably shouldn't be capitalized

File: lints/lint_ian_dns_name_includes_null_char_test.go [hunk 9]
```diff
@@ -1,12 +1,12 @@
-// lint_ian_dns_name_includes_null_char_test.go
+// lint_IAN_dns_name_includes_null_char_test.go
```
Comment (zakird) [inline lints/lint_ian_dns_name_includes_null_char_test.go]: shouldn't be capitalized

Comment (zhengping12) [inline lints/lint_ian_dns_name_includes_null_char_test.go]: fixed

File: lints/lint_ian_dns_name_includes_null_char.go (hunk 8 above)
Comment (zhengping12) [inline lints/lint_ian_dns_name_includes_null_char.go]: fixed

File: lints/lint_name_constraint_on_edi_party_name.go [hunk 10]
```diff
@@ -19,19 +19,19 @@ import (
 	"github.com/zmap/zlint/util"
 )
 
-type ncOnEdi struct {
+type ncOnEDI struct {
```
Comment (zakird) [inline lints/lint_name_constraint_on_edi_party_name.go]: you might spell out nc as name constraint. That's not an acronym that's generally used in this space.

File: lints/lint_ext_san_critical_with_subject_dn.go (hunk 7 above)
Comment (zhengping12) [inline lints/lint_ext_san_critical_with_subject_dn.go]: fixed

File: lints/lint_ext_san_edi_party_name_present_test.go (hunk 6 above)
Comment (zhengping12) [inline lints/lint_ext_san_edi_party_name_present_test.go]: fixed

File: lints/lint_ext_ian_no_entries.go (hunk 2 above)
Comment (zhengping12) [inline lints/lint_ext_ian_no_entries.go]: fixed.

File: lints/lint_ext_ian_dns_not_ia5_string_test.go (hunk 1 above)
Comment (zhengping12) [inline lints/lint_ext_ian_dns_not_ia5_string_test.go]: fixed

**Comment by zakird** [comment]
Looks fine to me.

**Comment by zakird** [comment]
Well. Except that tests don't pass! 😝

**Comment by zhengping12** [comment]
@zakird :( one of the spellings broke the test. But it should be all good now...

File: lints/lint_name_constraint_on_edi_party_name.go (hunk 10 above)
Comment (zhengping12) [inline lints/lint_name_constraint_on_edi_party_name.go]: fixed.

**Comment by zakird** [comment]
No worries. Once tests pass, I'll merge.

**Pull request merged**